feat(nepserver): add dnsserver to nepserver

wrenix opened the pull request:
Fix #7

Sn0w3y committed:
Integrate DNS Server functionality

Sn0w3y committed:
Update nepserver.py

Sn0w3y committed:
Update nepserver.py

Sn0w3y committed:
Update nepserver.py

wrenix commented on nepserver.py:
```diff
@@ -1,13 +1,43 @@
+        if str(request.q.qname) == INTERCEPT_DOMAIN:
+            # Intercept specific domain and respond with specified IP
+            reply.add_answer(RR(request.q.qname, QTYPE.A, rdata=A(RESPONSE_IP), ttl=60))
+        else:
```
make it possible to disable Forward
```suggestion
        else if FORWARD_DNS not in ["", "false", "False"]:
```

wrenix commented on nepserver.py:
```diff
@@ -141,5 +171,15 @@ def handler(*args):
         print("Server stopped.")
 
+def run_dns_server():
+    with socketserver.ThreadingUDPServer(('0.0.0.0', 53), DNSRequestHandler) as server:
```
Please make the listen ip configurable (and maybe an option to disable DNS).

wrenix reviewed (commented):
Looks good for the first version, i like to have 2-3 more options

Sn0w3y committed:
Update nepserver.py

Sn0w3y commented:
@wrenix should be done now ?

Can you please test the functionability aswell?

Also yould you please improve the Server How to ?

wrenix commented on nepserver.py:
```diff
@@ -15,10 +15,14 @@
 
 class DNSRequestHandler(socketserver.BaseRequestHandler):
     def handle(self):
+        if DISABLE_DNS:
```
not needed it is already disabled in:
`if __name__ == "__main__":`

wrenix commented:
i will test it in the evening

Sn0w3y committed:
Update nepserver.py

Sn0w3y commented:
> i will test it in the evening, if it works for you - merge it and i write an bugfix (if i found a problem).

fixed the "double check" please test it as i did not test it yet :D

Sn0w3y committed:
Update main.py

Class Definition Order: Moving the Datapoint class definition to the top makes the script more organized and easier to read. It's a good practice to define classes before they are used in the script.

Correct Property Use: The corrected implementation of property decorators for serial_number and watt ensures that these are properly handled as instance variables. This avoids the recursion error and makes the code work as expected.

URL Construction: Using urljoin for constructing the URL is a safer and more reliable method, especially when dealing with base URLs that might not always end with a slash.

Main Function: Encapsulating the main execution logic in a main function and protecting it with an if __name__ == "__main__": block is a best practice. It prevents the code from being executed when the script is imported as a module in another script.

Improved Print Statements: The print statements are clearer and provide better feedback on the script's execution progress.

Error Handling and Readability: While explicit error handling (e.g., try-except blocks) isn't added in this snippet, the overall structure of the code is cleaner, which makes adding error handling and debugging easier in the future.

wrenix commented:
please also/share the same Datapoint class on the `nepserver.py` then `main.py` (fix in nepserver and remove the old models.py)

Commit:
fix(nepserver): use Datapoint from main and move enviroment variables

Commit:
fix(nepserver): rename some enviroment variables

wrenix commented:
locks like the dns part works like expacted, i will validate with the next startup of the next inverter

Sn0w3y commented:
If validated we can Merge it :) also i guess we still need to check because sometimes we get errors if defining the same IP Addresses in the variables (which makes sense)

wrenix commented:
this code works for me, i would like to merge that.

What do you mean with same IP-Address?
(Every component listen on other Ports, so it should work)
I rename some environments variables and DNS- and HTTP-Server could run on different ip-address now.
